cmd/atlas/internal/migrate/migrate.go

```diff
@@ -257,6 +265,21 @@ func (r *EntRevisions) Migrate(ctx context.Context) (err error) {
 	}))
 }
 
+// maySetSchemaQualifier sets the schema on the atlas_schema_revisions table
+// for cases the migration should use qualified identifiers due to some limitation
+// in PostgreSQL services that use "connection pooler" and do not support the "search_path"
+// parameter. See: https://github.com/ariga/atlas/issues/2509.
+func (r *EntRevisions) maySetSchemaQualifier(t *schema.Table) {
```

Comment:
add test?

Reply:
Mocking the `database/sql/driver` in unit tests is really cumbersome (I just tried it), and in integration tests, it requires changing `/etc/hosts`.  

It's not really worth it. I'll let it run this way for some time, and then we can test it on all PGs.